In [27]:
import pandas as pd
import random
pd.set_option("display.max_rows", None)

In [28]:
df_msong = pd.read_csv("msongs.csv") #read msong.csv(1990-2010)
df_rank = pd.read_csv("ranking.csv") #read ranking.csv(1956-2019)
df_msong = df_msong.drop("Unnamed: 0", axis=1) #drop the column of "Unnamed: 0"
df_rank = df_rank.drop("Unnamed: 0", axis=1) #drop the column of "Unnamed: 0"

In [29]:
df_msong.year.value_counts()

2006    885
2007    864
2005    842
2008    765
2009    721
2004    646
2003    610
2002    504
2001    501
1999    475
2000    468
1998    407
1997    343
1995    338
1996    319
1994    260
1993    254
1992    226
2010    221
1991    187
1990    164
Name: year, dtype: int64

In [30]:
df_rank = df_rank[(df_rank["year"] >= 1990) & (df_rank["year"] <= 2010)]

In [31]:
dfm = pd.read_csv("audio_features_msongs.csv") #7655
dfa = pd.read_csv("audio_features.csv") #5658(all the year)
dfa = dfa[(dfa["year"] >= 1990) & (dfa["year"] <= 2010)] #1842(1990-2010)

#drop the unnecessary column
dfm=dfm.drop("Unnamed: 0", axis=1)
dfa=dfa.drop("Unnamed: 0", axis=1)

#drop duplicated rows
dfm = dfm.drop_duplicates(subset=['id'], keep='first')#7631
dfa = dfa.drop_duplicates(subset=['id'], keep='first') #1707

In [32]:
print(len(dfm))
print(len(dfa))

7631
1707


"dfm" is the audio_features data from 1990-2010. "dfa" is the audio_features data from 1990-2010 on bilboard top100.

In [33]:
#remove same elements as dfa from dfm
c=0

for i in dfm.id:
    for i2 in dfa.id:
        if(i == i2):
            
            a = dfm[dfm.id == i]
            dfm=dfm.drop(index = a.index)            

In [34]:
print(len(dfm))
print(len(dfa))

7583
1707


The above means there are 48 top100 bilboard(bbd) songs in the data, then we removed them.

In [35]:
dfm["bbd"] = 0
dfa["bbd"] = 1

In [36]:
#ran_list = random.sample(list(dfa.index), 100) #get the random list

#make the df. len is 7706, with the 100 top 100 songs, while 7606 songs are outside top100. 
#for i in ran_list:
#    dfm = dfm.append(dfa.loc[[i]], ignore_index=True)

In [41]:
dfm = dfm.append(dfa, ignore_index=True)

In [43]:
print(len(dfm))

9290


The above means we add the 1707 top100 songs which is 7580+1701=9290. Now, dfm has 1701 top100 songs and 7583 other songs.

In [44]:
dfm.to_csv("t.csv")

Use this for training and test the machine learning